urlc98c3b97d2_Trimax_venta_tgz-pJ8-com_gesoftware_venta_structures_map_MapTestJ8/src/com/gesoftware/venta/structures/map/RemoteMap.java

```diff
@@ -0,0 +1,432 @@
+                m_Get = new Option<Object>(object.m_Value);
+                notifyEveryOne();
+                return null;
+            }
+        });
+
+        /* Put dispatcher */
+        m_Puppeteer.registerDispatcher(ResponsePut.class, new IDispatcher<ResponsePut>() {
+            @Override
+            public final Serializable dispatch(final ResponsePut object) {
+                m_Put = new Option<Object>(object.m_Value);
+                notifyEveryOne();
+                return null;
+            }
+        });
+
+        /* Remove dispatcher */
+        m_Puppeteer.registerDispatcher(ResponseRemove.class, new IDispatcher<ResponseRemove>() {
+            @Override
+            public final Serializable dispatch(final ResponseRemove object) {
+                m_Remove = new Option<Object>(object.m_Value);
+                notifyEveryOne();
+                return null;
+            }
+        });
+
+        /* Key set dispatcher */
+        m_Puppeteer.registerDispatcher(ResponseKeySet.class, new IDispatcher<ResponseKeySet>() {
+            @Override
+            public final Serializable dispatch(final ResponseKeySet object) {
+                m_KeySet = object.m_Keys;
+                notifyEveryOne();
+                return null;
+            }
+        });
+
+        /* Values dispatcher */
+        m_Puppeteer.registerDispatcher(ResponseValues.class, new IDispatcher<ResponseValues>() {
+            @Override
+            public final Serializable dispatch(final ResponseValues object) {
+                m_Values = object.m_Values;
+                notifyEveryOne();
+                return null;
+            }
+        });
+
+        /* Entry set dispatcher */
+        m_Puppeteer.registerDispatcher(ResponseEntrySet.class, new IDispatcher<ResponseEntrySet>() {
+            @Override
+            public final Serializable dispatch(final ResponseEntrySet object) {
+                m_EntrySet = object.m_EntrySet;
+                notifyEveryOne();
+                return null;
+            }
+        });
+    } /* End of 'RemoteMap::registerDispatchers' method */
+
+    @Override
+    public final int size() {
+        m_Puppeteer.giveCommand(new QuerySize());
+
+        synchronized (this) {
+            try {
+                while (m_Size == null)
+                    wait();
+            } catch (final InterruptedException e) {
+                LoggingUtility.error("Can't wait: " + e.getMessage());
+            }
+        }
+
+        if (m_Size == null)
+            return 0;
+
+        final int size = m_Size.get();
+        m_Size = null;
+        return size;
+    } /* End of 'RemoteMap::size' method */
+
+    @Override
+    public boolean isEmpty() {
+        m_Puppeteer.giveCommand(new QueryIsEmpty());
+
+        synchronized (this) {
+            try {
+                while (m_IsEmpty == null)
+                    wait();
+            } catch (final InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)